python/dgl/data/csv_dataset.py

```diff
@@ -141,14 +144,21 @@ def load(self):
         graph_path = os.path.join(self.save_path,
                                   self.name + '.bin')
         self.graphs, self.data = load_graphs(graph_path)
+        if len(self.data) == 1:
+            self.labels = list(self.data.values())[0]
 
     def __getitem__(self, i):
+        if F.is_tensor(i) and F.ndim(i) == 1:
+            return Subset(self, F.copy_to(i, F.cpu()))
+
         if self._transform is None:
             g = self.graphs[i]
         else:
             g = self._transform(self.graphs[i])
 
-        if len(self.data) > 0:
+        if len(self.data) == 1:
```

Comment:
after this change, if we have single key in `self.data`, such key will be returned directly such as `label` or `train_mask`. If we have multiple keys in `self.data` such as `label` and `train_mask`, then `label` will not be returned directly. Right? all do we want to always return `label` if it exits?

Reply:
> such key will be returned directly such as label or train_mask

The value associated with the key will be returned directly. Mostly this will be the case for labels.

> If we have multiple keys in self.data such as label and train_mask, then label will not be returned directly. Right? all do we want to always return label if it exits?

My implementation is the former case.